pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -0,0 +1,763 @@
+			volumeToMount.VolumeName,
+			volumeToMount.VolumeSpec.Name(),
+			volumeToMount.PodName,
+			volumeToMount.Pod.UID,
+			err)
+	}
+
+	volumeMounter, newMounterErr := mountableVolumePlugin.NewMounter(
+		volumeToMount.VolumeSpec,
+		volumeToMount.Pod,
+		volume.VolumeOptions{})
+	if newMounterErr != nil {
+		return nil, fmt.Errorf(
+			"MountVolume.NewMounter failed for volume %q (spec.Name: %q) pod %q (UID: %q) with: %v",
+			volumeToMount.VolumeName,
+			volumeToMount.VolumeSpec.Name(),
+			volumeToMount.PodName,
+			volumeToMount.Pod.UID,
+			newMounterErr)
+	}
+
+	// Get attacher, if possible
+	attachableVolumePlugin, _ :=
+		oe.volumePluginMgr.FindAttachablePluginBySpec(volumeToMount.VolumeSpec)
+	var volumeAttacher volume.Attacher
+	if attachableVolumePlugin != nil {
+		volumeAttacher, _ = attachableVolumePlugin.NewAttacher()
+	}
+
+	var fsGroup *int64
+	if volumeToMount.Pod.Spec.SecurityContext != nil &&
+		volumeToMount.Pod.Spec.SecurityContext.FSGroup != nil {
+		fsGroup = volumeToMount.Pod.Spec.SecurityContext.FSGroup
+	}
+
+	return func() error {
+		if volumeAttacher != nil {
+			// Wait for attachable volumes to finish attaching
+			glog.Infof(
+				"Entering MountVolume.WaitForAttach for volume %q (spec.Name: %q) pod %q (UID: %q).",
+				volumeToMount.VolumeName,
+				volumeToMount.VolumeSpec.Name(),
+				volumeToMount.PodName,
+				volumeToMount.Pod.UID)
+
+			devicePath, err := volumeAttacher.WaitForAttach(
+				volumeToMount.VolumeSpec, waitForAttachTimeout)
+			if err != nil {
+				glog.Errorf(
+					"MountVolume.WaitForAttach failed for volume %q (spec.Name: %q) pod %q (UID: %q) with: %v",
+					volumeToMount.VolumeName,
+					volumeToMount.VolumeSpec.Name(),
+					volumeToMount.PodName,
+					volumeToMount.Pod.UID,
+					err)
+				return err
+			}
+
+			glog.Infof(
+				"MountVolume.WaitForAttach succeeded for volume %q (spec.Name: %q) pod %q (UID: %q).",
+				volumeToMount.VolumeName,
+				volumeToMount.VolumeSpec.Name(),
+				volumeToMount.PodName,
+				volumeToMount.Pod.UID)
+
+			// Mount device to global mount path
+			err = volumeAttacher.MountDevice(
+				volumeToMount.VolumeSpec,
+				devicePath,
+				volumeAttacher.GetDeviceMountPath(volumeToMount.VolumeSpec))
+			if err != nil {
+				glog.Errorf(
+					"MountVolume.MountDevice failed for volume %q (spec.Name: %q) pod %q (UID: %q) with: %v",
+					volumeToMount.VolumeName,
+					volumeToMount.VolumeSpec.Name(),
+					volumeToMount.PodName,
+					volumeToMount.Pod.UID,
+					err)
+				return err
+			}
+
+			glog.Infof(
+				"MountVolume.MountDevice succeeded for volume %q (spec.Name: %q) pod %q (UID: %q).",
+				volumeToMount.VolumeName,
+				volumeToMount.VolumeSpec.Name(),
+				volumeToMount.PodName,
+				volumeToMount.Pod.UID)
+
+			// Update actual state of world to refelct volume is globally mounted
+			markDeviceMountedErr := actualStateOfWorld.MarkDeviceAsMounted(
```

Comment:
`MountDevice()` is a method of the `Attacher` interface. "Non-attacher" plugins get mounted by `volumeMounter.SetUp()`. Arguably, `Attacher.MountDevice` could be eliminated and `SetUp()` could take on the responsibility of mounting the device and bind mounting it to the pods. But we can consider that outside of this PR.